kivy/uix/textinput.py

```diff
@@ -1593,27 +1594,27 @@ def _show_cut_copy_paste(self, pos, win, parent_changed=False,
         win_size = win.size
         bubble_pos = (t_pos[0], t_pos[1] + inch(.25))
 
-        if (bubble_pos[0] - bubble_hw) < 0:
+        if (bubble_pos[0] - bubble_hw + self.x) < 0:
```

Comment:
@akshayaurora got your point about half-visiblity but when it is off-screen (not on the screen)  then  no touch is made on the textinput so anyway no bubble is going to appear.